logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java

```diff
@@ -294,12 +310,58 @@ private boolean isOutput(final Vertex vertex) {
         return outputs.containsKey(vertex.getId());
     }
 
+    public final class CompiledOrderedExecution extends CompiledExecution {
+
+        @Override
+        public void compute(final QueueBatch batch, final boolean flush, final boolean shutdown) {
+           compute(batch.collection(), flush, shutdown);
+        }
+
+        @SuppressWarnings({"rawtypes"})
+        @Override
+        public void compute(final RubyArray batch, final boolean flush, final boolean shutdown) {
+            compute(batch, flush, shutdown);
+        }
+
+        @SuppressWarnings({"rawtypes", "unchecked"})
+        private void compute(final Collection batch, final boolean flush, final boolean shutdown) {
+            final RubyArray<RubyEvent> outputBatch = RubyUtil.RUBY.newArray();
+            // send batch one-by-one as single-element batches down the filters
+            final RubyArray<RubyEvent> filterBatch = RubyUtil.RUBY.newArray(1);
+            batch.forEach((e) -> {
```

Comment:
Is it useful to have `forEach` invocation instead of plain for loop?

Reply:
nope, will change.